test/startservers.py

```diff
@@ -45,7 +45,7 @@ def install(race_detection):
 
 def run(cmd, race_detection):
     # Note: Must use exec here so that killing this process kills the command.
-    cmd = """GORACE="halt_on_error=1" exec ./bin/%s""" % cmd
+    cmd = """GORACE="halt_on_error=1" PKCS11_PROXY_SOCKET="tcp://boulder-hsm:5657" exec ./bin/%s""" % cmd
```

Comment:
Passing two env vars on the command line like this is starting to get a bit messy. Last time I tried to use the `env` param here I ran into trouble because it overrides the default env. Can you experiment with making a copy of the default env, adding the vars we want, and passing that?